core/src/main/scala/org/http4s/rho/AuthInfo.scala

```diff
@@ -0,0 +1,18 @@
+package org.http4s.rho
+
+/** Typeclass for authentication embedding and retrieval from [[org.http4s.Request]] attributes map
+  *
+  * @tparam U Authentication info type for ex: User(id: UUID...)
+  */
+trait AuthInfo[U] {
```

Comment:
Why do we need to serialize and then deserialize the type `U` when stuffing `U` in and pulling it out of the attribute map? Why not just put type `U` in there?